src/main/java/seedu/address/model/VersionedEventStorage.java

```diff
@@ -0,0 +1,120 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.event.Event;
+import seedu.address.model.event.UniqueEventList;
+
+/**
+ * Stores the history of Events.
+ */
+public class VersionedEventStorage extends EventStorage {
+    private final ArrayList<UniqueEventList> versionedEvents = new ArrayList<>();
```

Comment:
bro how did u think of this

Reply:
my 2nd name is einstein

Reply:
jk they gave some general directions in AB4

Reply:
actl damn smart